presto-main/src/main/java/io/prestosql/server/security/CertificateAuthenticatorManager.java

```diff
@@ -0,0 +1,85 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.prestosql.spi.security.CertificateAuthenticator;
+import io.prestosql.spi.security.CertificateAuthenticatorFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static io.prestosql.util.PropertiesUtil.loadProperties;
+import static java.util.Objects.requireNonNull;
+
+public class CertificateAuthenticatorManager
+{
+    private static final Logger log = Logger.get(CertificateAuthenticatorManager.class);
+
+    private static final File CONFIG_FILE = new File("etc/certificate-authenticator.properties");
+    private static final String NAME_PROPERTY = "certificate-authenticator.name";
+
+    private final AtomicBoolean required = new AtomicBoolean();
+    private final Map<String, CertificateAuthenticatorFactory> factories = new ConcurrentHashMap<>();
+    private final AtomicReference<CertificateAuthenticator> authenticator = new AtomicReference<>();
+
+    public void setRequired()
+    {
+        required.set(true);
+    }
+
+    public void addCertificateAuthenticatorFactory(CertificateAuthenticatorFactory factory)
+    {
+        checkArgument(factories.putIfAbsent(factory.getName(), factory) == null,
+                "Certificate authenticator '%s' is already registered", factory.getName());
+    }
+
+    public void loadCertificateAuthenticator()
+            throws Exception
+    {
+        if (!required.get()) {
+            return;
+        }
+
+        File configFile = CONFIG_FILE.getAbsoluteFile();
+        Map<String, String> properties = new HashMap<>(loadProperties(configFile));
+
+        String name = properties.remove(NAME_PROPERTY);
+        checkState(!isNullOrEmpty(name), "Certificate authenticator configuration %s does not contain '%s'", configFile, NAME_PROPERTY);
+
+        log.info("-- Loading certificate authenticator --");
+
+        CertificateAuthenticatorFactory factory = factories.get(name);
+        checkState(factory != null, "Certificate authenticator '%s' is not registered", name);
+
+        CertificateAuthenticator authenticator = factory.create(ImmutableMap.copyOf(properties));
+        this.authenticator.set(requireNonNull(authenticator, "authenticator is null"));
+
+        log.info("-- Loaded certificate authenticator %s --", name);
+    }
+
+    public CertificateAuthenticator getAuthenticator()
```

Comment:
This is not used, which means the plugin code is actually never called. Did the change to `io.prestosql.server.security.CertificateAuthenticator` get dropped in this new PR?

Reply:
yes, probably an issue while rebasing and squashing the last time, i have fixed it now